app/browse-posts/index.tsx

```diff
@@ -0,0 +1,290 @@
+                        const data = {
+                            id: itemDoc.id,
+                            ...itemDoc.data(),
+                        } as ItemType;
+                        newItems.push(data);
+                    });
+
+                    for (const postId of postsToFetch) {
+                        newItemsByPost[postId] = newItems.filter(
+                            (item) => item.postId === postId,
+                        );
+                    }
+
+                    setItemsByPost((prev) => ({ ...prev, ...newItemsByPost }));
+                },
+                (error) => {
+                    console.error(error);
+                },
+            );
+
+            setItemUnsubscribers((prev) => [...prev, unsubscribeItems]);
+        } catch (error) {
+            console.error(error);
+        }
+    }
+
+    useEffect(() => {
+        const usersQuery = query(collection(db, "users"));
+        const unsubscribeUsers = onSnapshot(
+            usersQuery,
+            (usersQuerySnapshot) => {
+                const fetchedUsers: UserDataType[] = [];
+                usersQuerySnapshot.forEach((userDoc) => {
+                    const data = { id: userDoc.id, ...userDoc.data() };
+                    fetchedUsers.push(data as UserDataType);
+                });
+                setUsers(fetchedUsers);
+            },
+            (error) => {
+                console.error(error);
+            },
+        );
+
+        return () => unsubscribeUsers();
+    }, []);
+
+    function getUserById(userId: string) {
+        return users.find((user) => user.id === userId);
+    }
+
+    const onViewableItemsChanged = useCallback(
+        ({ viewableItems }: { viewableItems: ViewToken[] }) => {
+            const visibleIds = viewableItems
+                .map((item) => (item.item as PostType).id)
+                .filter((id): id is string => id !== undefined);
+
+            setVisiblePostIds(visibleIds);
+        },
+        [],
+    );
+
+    interface RenderItemProps {
+        item: PostType;
+        index: number;
+    }
+
+    function renderItem(props: RenderItemProps) {
+        const { item: post, index } = props;
+        const postUser = getUserById(post.userId);
+        const postItems = itemsByPost[post.id] || [];
+        const isVisible = visiblePostIds.includes(post.id);
+
+        return (
+            <>
+                <View className="mb-5 flex flex-col">
+                    <PostHeader
+                        postId={post.id}
+                        userName={postUser?.name}
+                        userImageUrl={postUser?.imageUrl ?? ""}
+                    />
+                    <PostContent
+                        post={post}
+                        postItems={postItems}
+                        isVisible={isVisible}
+                    />
+                </View>
+                {index < posts.length - 1 && <PostDivider />}
+            </>
+        );
+    }
```

Comment:
Put this into its own component instead of a function

```tsx
renderItem={(item, index) => <PostItem item={item} .....rest_of_needed_props />}
```

Reply:
in here, i noticed nga may extracted component per little thing, like meron PostCaption tapos text lang inside HAHAHA, and yung PostDivider pero Divider lang inside, when mag create ka na nito na component `PostItem` can you minimize na ang mga extracted components? like if maliit lang man sila then no need na i separate component. maybe ang post-images or smth ang need lang i extract, thankyou!!